apps/builder/app/shared/copy-paste/plugin-instance.ts

```diff
@@ -159,9 +161,36 @@ const getTreeData = (targetInstanceSelector: InstanceSelector) => {
   // first item is guaranteed root of copied tree
   const treeInstances = getMapValuesByKeysSet(instances, treeInstanceIds);
 
+  const dataSources = dataSourcesStore.get();
   const treeProps = getMapValuesBy(propsStore.get(), (prop) =>
     treeInstanceIds.has(prop.instanceId)
-  );
+  ).map((prop) => {
+    // unbind data source from prop
+    if (prop.type === "dataSource") {
+      const dataSource = dataSources.get(prop.value);
+      if (dataSource?.type === "string") {
+        return {
+          id: prop.id,
+          instanceId: prop.instanceId,
+          name: prop.name,
+          type: dataSource.type,
+          value: dataSource.defaultValue,
+        } satisfies Prop;
+      }
+      if (dataSource?.type === "boolean") {
+        return {
+          id: prop.id,
+          instanceId: prop.instanceId,
+          name: prop.name,
+          type: dataSource.type,
+          value: dataSource.defaultValue,
+        } satisfies Prop;
+      }
+      // ensure all data source types are mapped to props
```

Comment:
these two look nearly identical, I think you might want to avoid typing these out since there will be more types and work this around with typescript in some different way

Reply:
Typescript does not allow to put them in one condition because type and defaultValue become two unions so we get multiplication of 4 types

cc @istarkov

Reply:
The issue is that https://www.typescriptlang.org/play?#code/C4TwDgpgBAIghsOBlA9gVwE4GNoF4oDeokAXFAEQDOwGAlgHYDm5ANFAG5wA2aEZ1dJgF8oAH0LE+FemgC2AIwgZWHbrzIyFSoQG4AUHqwp61KKlkQAYmnpYo+ABQATBHDLxEqTDgCU9gHyEelAhUBgQwJj0QaGxUJJkLogAdJIswXEhnDxSSXDJ2bwZIUJ6ukA

Gives all  combination of type and value
<img width="743" alt="image" src="https://github.com/webstudio-is/webstudio-builder/assets/5077042/9903e8c3-e5ca-401d-a7bb-5c5f89986d16">

Checking how to avoid

Reply:
That rare case when spread works like a charm
https://www.typescriptlang.org/play?#code/C4TwDgpgBAIghsOBlA9gVwE4GNoF4oDeokAXFAEQDOwGAlgHYDm5ANFAG5wA2aEZ1dJgF8oAH0LE+FemgC2AIwgZWHbrzIyFSoQG4AUJKgAFDCjBR8E8FKo0GzNpx5SB9tgAsIXLin53hYlak0nKKyo5qUpphHl4+foKMunp6WCj01FCoshAAYmj0WBZQABQAJghwZPCIqJg4AJQWAHyEelAdUGkZwFAYxYSdQ1Dtwx0AdJMViCyjY57evlAA5ACMy7NDyUMYEMCY9H36ukA